opts/color.go

```diff
@@ -0,0 +1,117 @@
+package opts
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+)
+
+type Color string
+
+type ColorVar interface {
+	Text() Color
+}
+
+type rgb struct {
+	r uint16
+	g uint16
+	b uint16
+	a *float32
+}
+
+func (c rgb) Text() Color {
+	if c.a == nil {
+		return Color(fmt.Sprintf("rgb(%d, %d, %d)", c.r, c.g, c.b))
+	}
+	return Color(fmt.Sprintf("rgba(%d, %d, %d, %f)", c.r, c.g, c.b, *c.a))
+}
+
+func HexColor(str string) Color {
```

Comment:
I think it is redundant code here, Javascript supports hex color format naturally, hence why should we convert them into the RGB format?